src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelSchema.scala

```diff
@@ -14,35 +13,106 @@ import org.broadinstitute.dsde.firecloud.FireCloudConfig
+  def getPlural (entityType: String): Try[String]
+  def getRequiredAttributes(entityType: String): Try[Map[String, String]]
+  def getTypeSchema(entityType: String): Try[EntityMetadata]
+  def supportsBackwardsCompatibleIds(): Boolean
 
-  def getTypeSchema(entityType: String): Try[EntityMetadata] = {
-    EntityTypes.types.get(entityType) match {
-      case Some(schema) => Success(schema)
-      case None => Failure(new RuntimeException("Unknown entity type: " + entityType))
+  def isEntityTypeInSchema(entityType: String): Boolean = {
+    Try(this.memberTypeFromEntityType(entityType)) match {
+      case Failure(_) => false
+      case _ => true
     }
   }
+}
+
+
+
+object SchemaTypes {
+  sealed trait SchemaType
+  final case object FIRECLOUD extends SchemaType { override def toString = "firecloud" }
+  final case object FLEXIBLE extends SchemaType { override def toString = "flexible" }
 
-  def isCollectionType(entityType: String): Try[Boolean] = {
-    getTypeSchema(entityType).map(_.memberType.isDefined)
+  def withName(name: String): SchemaType = {
+    name.toLowerCase match {
+      case "firecloud" => FIRECLOUD
+      case "flexible" => FLEXIBLE
+      case _ => throw new FireCloudException(s"Invalid schema type '$name', supported types are: firecloud, flexible")
+    }
   }
+}
+
+
+
+object ModelSchemaRegistry {
+  // add new schema types from most specific to most general
+  val schemas: Map[SchemaTypes.SchemaType, ModelSchema] = Map(SchemaTypes.FIRECLOUD -> new FirecloudModelSchema, SchemaTypes.FLEXIBLE -> new FlexibleModelSchema)
+
+  def getModelForSchemaType(schemaType: SchemaTypes.SchemaType): ModelSchema = schemas.getOrElse(schemaType, schemas.last._2)
 
-  def getCollectionMemberType(collectionEntityType: String): Try[Option[String]] = {
-    getTypeSchema(collectionEntityType).map(_.memberType)
+  def getModelForEntityType(entityType: String): ModelSchema = {
+    val possibleMatch = schemas.values.collectFirst({case schema if schema.isEntityTypeInSchema(entityType) => schema})
+    possibleMatch.getOrElse(schemas.last._2)
   }
+}
 
-  def getPlural(entityType: String): Try[String] = {
-    getTypeSchema(entityType).map(_.plural)
+
+class FlexibleModelSchema extends ModelSchema {
+
+  def memberTypeFromEntityType(entityType: String): Try[Option[String]] = {
+    Success(Some(entityType.replace("_set", "")).filter(_ => isCollectionType(entityType)))
+  }
+
+  def isCollectionType(entityType: String): Boolean = {
+    entityType.endsWith("_set")
   }
 
   def getRequiredAttributes(entityType: String): Try[Map[String, String]] = {
-    getTypeSchema(entityType).map(_.requiredAttributes)
+    Success(Map.empty)
   }
+
+  def getPlural(entityType: String): Try[String] =  {
+    Success(entityType + "s")
+  }
+
+  def getTypeSchema(entityType: String): Try[EntityMetadata] = {
+    Success(EntityMetadata(entityType, Map.empty, Some(entityType+"_members").filter(_ => isCollectionType(entityType))))
```

Comment:
Not sure if it matters right now but the first property of `EntityMetadata` should be the pluralized entity type. I suggest extracting a function from the current `getPlural`:
```
  private def pluralize(entityType: String) = {
    entityType + "s"
  }
```